src/contracts/core/AVSDirectory.sol

```diff
@@ -127,20 +337,61 @@ contract AVSDirectory is
     }
 
     /**
-     * @notice Called by an avs to emit an `AVSMetadataURIUpdated` event indicating the information has updated.
-     * @param metadataURI The URI for metadata associated with an avs
+     *
+     *                         INTERNAL FUNCTIONS
+     *
      */
-    function updateAVSMetadataURI(string calldata metadataURI) external {
-        emit AVSMetadataURIUpdated(msg.sender, metadataURI);
+
+    /**
+     * @notice Helper function used by migration & registration functions to register an operator to operator sets.
+     * @param avs The AVS that the operator is registering to.
+     * @param operator The operator to register.
+     * @param operatorSetIds The IDs of the operator sets.
+     */
+    function _registerToOperatorSets(address operator, address avs, uint32[] calldata operatorSetIds) internal {
+        // Loop over `operatorSetIds` array and register `operator` for each item.
+        for (uint256 i = 0; i < operatorSetIds.length; ++i) {
+            OperatorSet memory operatorSet = OperatorSet(avs, operatorSetIds[i]);
+
+            require(isOperatorSet[avs][operatorSetIds[i]], InvalidOperatorSet());
+
+            bytes32 encodedOperatorSet = _encodeOperatorSet(operatorSet);
+
+            require(_operatorSetsMemberOf[operator].add(encodedOperatorSet), InvalidOperator());
+
+            _operatorSetMembers[encodedOperatorSet].add(operator);
```

Comment:
is there a reason for this asymmetry?
it feels like the second line should also be wrapped in a require statement -- am I missing something?

Reply:
ah, at least it seems these are always meant to mirror one another, so if the first addition returns 'true' (indicating the addition occurred) then the second should also return 'true', and if the first returns 'false' then execution has already reverted